src/utils/settings.ts

```diff
@@ -0,0 +1,39 @@
+import { DEFAULT_GAS_PRICE } from 'config'
+
+export interface UserSettings {
+  gasPrice: string
+}
+
+export const VERSION = 1
+
+export const GAS_SETTINGS = {
+  default: DEFAULT_GAS_PRICE, // 5
+  fast: '10000000000', // 10
+  reallyfast: '15000000000', // 15
+}
+
+export const getDefaultSettings = (): UserSettings => ({
+  gasPrice: GAS_SETTINGS.default,
+})
+
+export const getStorageKey = (account: string) => {
+  return `pancakeswap_settings_${account}_${VERSION}`
+}
+
+export const getSettings = (account: string): UserSettings => {
+  try {
+    const settingsFromLs = localStorage.getItem(getStorageKey(account))
```

Comment:
Settings won't change often so I see no reason to put them in another state. This also allows for the interface to be framework agnostic.